frontend/dockerfile/dockerfile_test.go

```diff
@@ -586,6 +587,52 @@ Dockerfile
 	require.Equal(t, "bay-contents", string(dt))
 }
 
+func testDockerignoreInvalid(t *testing.T, sb integration.Sandbox) {
+	t.Parallel()
+
+	dockerfile := []byte(`
+FROM scratch
+COPY . .
+`)
+
+	dir, err := tmpdir(
+		fstest.CreateFile("Dockerfile", dockerfile, 0600),
+		fstest.CreateFile(".dockerignore", []byte("!\n"), 0600),
+	)
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	c, err := client.New(sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	destDir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(destDir)
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
you can do `context.WithTimeout` here and check that it has not returned after solve returns with error